packages/pigeon/run_tests.sh

```diff
@@ -387,7 +387,7 @@ if [ "$should_run_android_unittests" = true ]; then
   get_java_linter_formatter
 fi
 test_running_without_arguments
-if [ "$should_run_dart_unittests" = true ]; then
+if [ "$should_run_dart_unittests" = true ] && [ ! "$CIRRUS_CI" = true ]; then
```

Comment:
Embedding knowledge of CI environments into scripts tends to make understanding what's happening much harder. I would much prefer that the script be restructured so that the parts you want to be run by CI are in a script that's for that purpose, and the rest are in a different script, and if you want that to be transparent for people running tests manually you make the script that's designed for local running delegate as necessary.

Reply:
(I think this is exacerbated by the fact that CI is running something called "locas_tests.sh" in the first place; I don't know what "local" means if it's running on CI.)

Reply:
There are environment variables that are more generic, "$CI", if you'd like.  If the boolean expression fails, the failure is benign.

The idea for local_tests is that certain packages have more complex test harnesses, there isn't a generic wrapper for the whole repo that can run tests that have to do codegen and linting in different languages for example.  Plus we have the problem that we don't want CI running tests in a different way as people are developing locally.  We have to try to keep them as similar as possible.

I don't really think there is an alternative beyond just doing the same thing with Dart, but I'm open to ideas.

Reply:
I'm a little confused; most of my comment was a specific suggestion for how to structure it, so I'm not sure what you mean by "I don't really think there is an alternative". How is what I suggested not a viable alternative?

In fact, it doesn't even look like you need to go as far as what I suggested; you already have options for the tests, so you could literally just rename this script and then make `run_tests.sh` call that new script with options to disable the tests you don't want running in CI. Problem solved, no CI-specific knowledge necessary in the script.